text/0039-service-overhaul.md

```diff
@@ -0,0 +1,462 @@
+
+The notion of maximum and recommended shall be distinguished.
+Maximum, supported resolutions, recommended settings will be ignorable with their respective checkboxes.
+
+Those services should have no specific behavior like ingest management.
+
+But can list with which codec they are compatible if needed. [Thanks to RFC 45](https://github.com/tytan652/rfcs/blob/protocol_api/text/0045-protocols-in-outputs-api.md)
+
+In the end, adding streaming service will only be a JSON addition for this plugin.
+Only improvements will be accepted in the code of this plugin.
+
+#### A plugin per special case
+This/Those plugin(s) is/are meant to provide replacements for `rtmp_common` type for services which need a custom behavior.
+
+This/Those plugin(s) is/are meant to have implemented and register services which need a specific behavior like custom ingest with a unique id for each one.
+
+#### Note about the plugins
+- Like this if the service needs a specific behavior, you can 'transfer' it from `obs-services` to a new plugin and keep the same id. The change will be seamless for the user.
+
+### Conversion of services
+**Downgrade will break service configuration**
+A JSON or a harcoded list with old service name linked to their new id, to make OBS able to convert the `service.json` to a new one.
+
+Auth integration config inside `basic.ini` will be also transfered to the new file.
+
+## Other changes
+
+### Service API changes
+- Add missing get_properties2 and get_default2 functions for service
+
+### Property API addition
+- Add a property to be able to show information with text label
+
+Since the service API is in a way unusable by third-party, any breakage will only impact OBS built-in service plugins ( so only `rtmp-services`).
+
+### Service integrations new flow
+
+#### How docks and CEF widget for OAuth will be created
+For the docks we could add the possibility to add them through the frontend API.
+
+Adding to the frontend API the possibility to generate the browser widget neutral on the system use (CEF). Also need to add a function asking OBS Studio is browser features are availlable.
+
+OR
+
+Maybe adding an API to `obs-browser` to allow plugins to create widget and docks it without using the frontend API as a bridge.
+
+#### How docks will be added
+
+`OBS_FRONTEND_EVENT_FINISHED_LOADING` can be used when OBS startup to enable docks.
+
+Three new front-end event will be added:
+- `OBS_FRONTEND_EVENT_SERVICE_CHANGED` emitted when the service was changed (new id).
+- `OBS_FRONTEND_EVENT_SERVICE_UPDATED` emitted when the service had a parameter updated (no id change).
+
+Each service integration plugin will react to at least the two first to add or remove his related docks.
+
+#### Manage Broadcast
+YouTube is not the only service that could have or need a "Manage Broadcast" button. So adding a way to make other service plugin able to use it.
```

Comment:
This sort of thing would definitely need to be part of a first version of this. Plus handling the Start Streaming button to intercept it in certain situations (such as opening Manage Broadcast indirectly).